csharp/extractor/Semmle.Extraction.CSharp/Entities/LocalVariable.cs

```diff
@@ -9,10 +10,12 @@ class LocalVariable : CachedSymbol<ISymbol>
 
         public override void WriteId(TextWriter trapFile)
         {
-            trapFile.WriteSubId(Location);
-            trapFile.Write('_');
-            trapFile.Write(symbol.Name);
-            trapFile.Write(";localvar");
+            throw new InvalidOperationException();
```

Comment:
I think there was a good reason for not using a fresh entity, but I can't exactly remember why (I think it may have to do with cases where a local variable entity is created before the corresponding local variable declaration). How about instead [doing the same](https://github.com/Semmle/ql/blob/master/csharp/extractor/Semmle.Extraction.CSharp/Entities/LocalFunction.cs#L16) as for local functions, i.e., prepend the enclosing type (which will contain the assembly info)?

Reply:
Need to do the same fix for local functions that could also become inconsistent when extracted twice.